framework/src/mfem/functormaterials/MFEMGenericFunctorVectorMaterial.C

```diff
@@ -5,6 +5,59 @@
 
 registerMooseObject("MooseApp", MFEMGenericFunctorVectorMaterial);
 
+/// Handle any numerical vector values, which should be enclosed in curly braces
+std::vector<MFEMVectorCoefficientName>
+processLiteralVectors(const std::vector<MFEMVectorCoefficientName> & input)
+{
+  std::vector<MFEMVectorCoefficientName> result;
+  bool in_literal = false;
+  MFEMVectorCoefficientName literal;
+  for (const auto & item : input)
+  {
+    if (in_literal)
+    {
+      if (item.front() == '{')
+      {
+        mooseError("Nested numeric vector values are not permitted in "
+                   "MFEMGenericFunctorVectoMaterial prop_values.");
+      }
+      else if (item.back() == '}')
+      {
+        in_literal = false;
+        literal += " " + item.substr(0, item.size() - 1);
+        result.push_back(literal);
+      }
+      else
+      {
+        literal += " " + item;
+      }
+    }
+    else if (item.front() == '{')
+    {
+      if (item.back() == '}')
+      {
+        result.push_back(item.substr(1, item.size() - 2));
```

Comment:
Why is it -2 here instead of -1 ?

Reply:
But earlier in the function you use `item.size() - 1` to remove a closing bracket

Reply:
Lol I even looked up the c++ reference for substr and didn't register this
> The second argument is not the final index but the length